pkg/trie/triedb/node_storage.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package triedb
+
+import (
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/gammazero/deque"
+)
+
+type StoredNode interface {
+	isStoredNode()
+	getNode() Node
+}
+
+type (
+	New struct {
+		node Node
+	}
+	Cached struct {
+		node Node
+		hash common.Hash
+	}
+)
+
+func (New) isStoredNode() {}
+func (n New) getNode() Node {
+	return n.node
+}
+func (Cached) isStoredNode() {}
+func (n Cached) getNode() Node {
+	return n.node
+}
+
+func NewNewNode(node Node) New {
```

Comment:
I get why this is this way, but potentially could change to `NewNodeNew` and `NewNodeCached`, up to you

Reply:
If we even need the cached, else ignore that part of the suggestion

Reply:
Yes I like your suggestion `NewNew` is confusing haha. I'll do the change